### PR TITLE
Add event querying and payment_intent events

### PR DESCRIPTION
**Pull request by bpcreech**
These features are technically separate but it's easier to test them together.

The Stripe event query API lets us filter by event type and timestamp.

Meanwhile payment_intent successes and failures generate events (this is particularly important for sepa_direct_debit which on the real Stripe API is typically pending for days, on the Stripe test API is typically pending for a minute or so, and in localstripe is pending for 0.5 seconds).

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -1068,6 +1068,12 @@ payment_intent=$(
        -d capture_method=manual \
   | grep -oE 'pi_\w+' | head -n 1)
 
+# we don't get a payment_intent.succeeded event from the pre-auth:
+succeded_event=$(
```
[the remark this answers is not in the document]
Comment (bpcreech) [reply test.sh:1080]: English is my first language. My high school literature teacher is spinning in her grave. (I don't think she passed, but she expressed some facets of vampirism.)

[the remark this answers is not in the document]
Comment (bpcreech) [reply test.sh]: done, thank you!

File: test.sh [hunk 2]
```diff
@@ -1068,6 +1068,12 @@ payment_intent=$(
...
+  curl -sSfg -u $SK: "$HOST/v1/events?type=payment_intent.succeeded" \
+  | grep -oE "^          \"id\": \"$payment_intent\"" || true)
```
[the remark this answers is not in the document]
Comment (bpcreech) [reply test.sh]: fixed!

I also removed a couple unnecessary `|| true`s... we only need those when the grep is expected to fail and I had two where when the grep is expected to succeed.

[the remark this answers is not in the document]
Comment (adrienverge) [reply test.sh]: Agreed, makes sense :+1:

File: test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrienverge) [reply test.sh]: :smile:

**Pull request merged**